src/test/java/org/jenkinsci/plugins/workflow/steps/ToolStepRunTest.java

```diff
@@ -60,4 +73,69 @@
         r.assertLogContains("Apache Maven 3", r.assertBuildStatusSuccess(p.scheduleBuild2(0)));
     }
 
+    @Test public void toolWithSymbol() throws Exception {
+        File buildDirectory = new File(System.getProperty("buildDirectory", "target")); // TODO relative path
+        File toolHome = new File(buildDirectory, "mockTools");
+        MockToolWithSymbol tool = new MockToolWithSymbol("mock-tool-with-symbol", toolHome.getAbsolutePath(), JenkinsRule.NO_PROPERTIES);
+        Jenkins.getInstance().getDescriptorByType(MockToolWithSymbol.MockToolWithSymbolDescriptor.class).setInstallations(tool);
+        String name = tool.getName();
+
+        String type = "mockToolWithSymbol";
+
+        WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p");
+        p.setDefinition(new CpsFlowDefinition("node {def home = tool name: '" + name + "', type: '" + type + "'}",
+                true));
+
+        r.assertBuildStatusSuccess(p.scheduleBuild2(0));
+    }
+
+    @Test public void toolWithoutSymbol() throws Exception {
+        File buildDirectory = new File(System.getProperty("buildDirectory", "target")); // TODO relative path
+        File toolHome = new File(buildDirectory, "mockTools");
+        MockToolWithoutSymbol tool = new MockToolWithoutSymbol("mock-tool-without-symbol", toolHome.getAbsolutePath(), JenkinsRule.NO_PROPERTIES);
+        Jenkins.getInstance().getDescriptorByType(MockToolWithoutSymbol.MockToolWithoutSymbolDescriptor.class).setInstallations(tool);
+        String name = tool.getName();
+
+        String type = "mockToolWithoutSymbol";
+
+        WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p");
+        p.setDefinition(new CpsFlowDefinition("node {def home = tool name: '" + name + "', type: '" + type + "'}",
+                true));
+
+        r.assertLogContains("No mockToolWithoutSymbol named mock-tool-without-symbol found",
```

Comment:
Should probably test the _normal_ case that you can specify the descriptor ID and have it work!